logic/data.py

```diff
@@ -53,6 +53,23 @@ def __init__(self, X: pd.DataFrame, y: NDArray):
             time_to_event=time_to_event,
         )
 
+    def drop_negative_and_zero_timepoints(self) -> int:
+        """
+        Drop rows with a negative or zero time to prepare data for log-transformation.
+
+        :return: Number of dropped rows
+        """
+        logging.info("Check data for negative and zero timepoints")
+        bad_rows = self.survival.time_to_event <= 0
```

Comment:
I think this would keep the rows where the survival time is e.g. a string.

Reply:
Non-numeric values should not be possible at this point. I suspect that text values that can not be interpreted as a numeric value will raise a `ValueError` in `logic.data.read_survival_data_np`. However, I don't have the test setup right now to verify this. Could you start a run with non-numeric values for the time, please?
Anyway, the preprocessing steps are a bit all over the place. I think I need to revise the data preprocessing part a bit to allow for good warning messages here.

Reply:
I will check it later, I think for now this is fine